sdk/identity/identity/src/msal/nodeFlows/msalClientCertificate.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { readFileSync } from "fs";
+import { createHash } from "crypto";
+import { AccessToken } from "@azure/core-http";
+import { MsalNodeOptions, MsalNode } from "./nodeCommon";
+import { formatError } from "../../util/logging";
+import { CredentialFlowGetTokenOptions } from "../credentials";
+
+/**
+ * Options that can be passed to configure MSAL to handle client certificates.
+ * @internal
+ */
+export interface MSALClientCertificateOptions extends MsalNodeOptions {
+  /**
+   * Location of the certificate.
+   */
+  certificatePath: string;
+  /**
+   * Option to include x5c header for SubjectName and Issuer name authorization.
+   * Set this option to send base64 encoded public certificate in the client assertion header as an x5c claim
+   */
+  sendCertificateChain?: boolean;
+}
+
+/**
+ * Parts of a certificate, as understood by MSAL.
+ * @internal
+ */
+interface CertificateParts {
+  /**
+   * Hex encoded X.509 SHA-1 thumbprint of the certificate
+   */
+  thumbprint: string;
+  /**
+   * The PEM encoded private key (string should contain -----BEGIN PRIVATE KEY----- ... -----END PRIVATE KEY-----
+   */
+  privateKey: string;
+  /**
+   * x5c header.
+   */
+  x5c: string;
+}
+
+/**
+ * MSAL client certificate client. Calls to MSAL's confidential application's `acquireTokenByClientCredential` during `doGetToken`.
+ * @internal
+ */
+export class MsalClientCertificate extends MsalNode {
+  private sendCertificateChain?: boolean;
+
+  constructor(options: MSALClientCertificateOptions) {
+    super(options);
+    this.requiresConfidential = true;
+    this.sendCertificateChain = options.sendCertificateChain;
+
+    const parts = this.parseCertificate(options.certificatePath);
+    this.msalConfig.auth.clientCertificate = parts;
+  }
+
+  private parseCertificate(certificatePath: string): CertificateParts {
+    const certificateParts: Partial<CertificateParts> = {};
+
+    certificateParts.privateKey = readFileSync(certificatePath, "utf8");
```

Comment:
Based off the parsing below it seems you expect the file to contain both the private key and public certs. Given that perhaps this property name is a bit confusing as it has all the contents not just the private key?